desktop/cmp/button/AppMenuButton.js

```diff
@@ -155,15 +156,30 @@ function buildMenuItems({
         }
     ];
 
-    return [
+    return parseMenuItems([
         ...extraItems,
         '-',
         ...defaultItems
-    ]
+    ]);
+}
+
+function parseMenuItems(items) {
+    return items
         .filter(it => !it.omit)
         .filter(filterConsecutiveMenuSeparators())
         .map(it => {
             if (it === '-') return menuDivider();
-            return isValidElement(it) ? it : menuItem(it);
+            if (isValidElement(it)) {
+                return ['Blueprint3.MenuItem', 'Blueprint3.MenuDivider'].includes(it.type.displayName) ?
+                    it :
+                    menuItem({text: it});
+            }
+
+            const cfg = {...it};
```

Comment:
This second case is expecting `it` will be a plain object, suitable as a config for `menuItem`, is that correct?

Can we tweak the proptype doc on `extraItems` for a bit more clarity? I think the valid options are:
- A `menuItem` or `menuDivider` element, which will be used as-is.
- A `ReactElement` or string (`isValidElement(someString) == true` ??) which will be set as the text of a menu item.
- A plain object to be used as a `menuItem` config, with support for a nested `items` array which can contain this same list of things.

Not saying we need to use those exact / verbose words in the prop-type doc, but enumerating them here to ensure I understand what we're accepting. At some point soon we really need to provide a first-class, fully documented general menu component, without relying on importing BP kit comps directly.  Still don't think we have an issue for that, will add.

Reply:
Yes, that's correct, and those are the options supported. I've pushed an update to the docs and propTypes for extra items that should hopefully clarify this a bit.